pinot-core/src/main/java/com/linkedin/pinot/core/query/pruner/columnvalue/ColumnValueSegmentPruner.java

```diff
@@ -0,0 +1,212 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.query.pruner.columnvalue;
+
+import com.linkedin.pinot.common.data.FieldSpec;
+import com.linkedin.pinot.common.request.BrokerRequest;
+import com.linkedin.pinot.common.request.FilterOperator;
+import com.linkedin.pinot.common.utils.request.FilterQueryTree;
+import com.linkedin.pinot.common.utils.request.RequestUtils;
+import com.linkedin.pinot.core.common.predicate.RangePredicate;
+import com.linkedin.pinot.core.indexsegment.IndexSegment;
+import com.linkedin.pinot.core.query.pruner.SegmentPruner;
+import com.linkedin.pinot.core.segment.index.ColumnMetadata;
+import com.linkedin.pinot.core.segment.index.SegmentMetadataImpl;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import org.apache.commons.configuration.Configuration;
+
+
+/**
+ * An implementation of SegmentPruner.
+ * <p>Pruner will prune segment based on the column value inside the filter.
+ */
+public class ColumnValueSegmentPruner implements SegmentPruner {
+
+  @Override
+  public void init(Configuration config) {
+  }
+
+  @Override
+  public boolean prune(@Nonnull IndexSegment segment, @Nonnull BrokerRequest brokerRequest) {
+    FilterQueryTree filterQueryTree = RequestUtils.generateFilterQueryTree(brokerRequest);
```

Comment:
This needs interface change. Will address this in another PR if this cause any performance issue.